rpcs3/Emu/System.cpp

```diff
@@ -408,6 +412,15 @@ void Emulator::Init(bool add_only)
 			}
 		}
 	}
+
+	if (!g_ipc_server && g_cfg.misc.ipc_server)
+	{
+		g_ipc_server = std::make_shared<named_thread<IPC_server>>();
+	}
+	else if (g_ipc_server && !g_cfg.misc.ipc_server)
+	{
+		g_ipc_server.reset();
+	}
```

Comment:
why is this done in all Init, Load, and Kill ?
Also, if you're copy pasting it three times, you should make it a function.

Reply:
I don't understand why not use FXO, if you explain the problem we may be able to orovide a solution. If conditional construction is the issue you can see how we manage native UI by using non-default constructor. If shared ptr really needs to be used, IDM can be used even for a single instance.

Reply:
You can create another default initialized FXO object that manages conditional construction through its constructor and destructor.

Reply:
For the record, about fxo, if your object has a non-default constructor, it will not be automatically constructed on boot and you can dynamically create it according to config. Only trivial constructor types have the automatic construction issue with named_thread.

Reply:
Decided to let the manager be alive always and have it manage the lifetime of the thread through a pointer. This should allow you to change the config on the fly and have the server shutdown and start up dynamically.